nengolib/neurons.py

```diff
@@ -5,7 +5,53 @@
 from nengo.builder.neurons import SimNeurons
 from nengo.neurons import NeuronType
 
-__all__ = ['PerfectLIF', 'Unit', 'Tanh']
+__all__ = ['PerfectLIF', 'Unit', 'Tanh', 'initialize_lif']
+
+
+def _sample_lif_state(sim, ens, x0, rng):
+    """Sample the LIF's voltage/refractory assuming uniform within ISI."""
+    lif = ens.neuron_type
+    params = sim.model.params
+
+    eval_points = x0[None, :]
+    x = np.dot(eval_points, params[ens].encoders.T / ens.radius)
+    a = ens.neuron_type.rates(x, params[ens].gain, params[ens].bias)
+    J = ens.neuron_type.current(x, params[ens].gain, params[ens].bias)
+
+    # fast-forward to a random time within the ISI for any active neurons
+    is_active = a > 0
+    t_isi = rng.rand(ens.n_neurons) * np.where(is_active, 1 / a, 0)
```

Comment:
NTS: This raises a warning for zero division, even though it's irrelevant because of the `np.where` selection. Do this with slicing instead.